pkg/agent/agent.go

```diff
@@ -234,11 +274,42 @@ func (a *Agent) newManager(ctx context.Context, sto storage.Storage, cat catalog
 	}
 
 	mgr := manager.New(config)
-	if err := mgr.Initialize(ctx); err != nil {
-		return nil, err
-	}
+	if a.c.RetryBootstrap {
+		initBackoffClock := clock.New()
+		initBackoff := backoff.NewBackoff(initBackoffClock, 5*time.Second, backoff.WithMaxElapsedTime(1*time.Minute))
+		for {
+			select {
+			case <-ctx.Done():
+				return nil, ctx.Err()
+			default:
+			}
+
+			err := mgr.Initialize(ctx)
+			if err == nil {
+				return mgr, nil
+			}
+
+			//TODO: what errors should make us return immediately?
```

Comment:
@amartinezfayo , in this case, should we verify for specific errors to return immediately?

Reply:
I think that we should check if we know that the agent should reattest or shutdown, in which case we should not retry. We have functions to know that based on the error. Something like:

```

			if nodeutil.ShouldAgentReattest(err) || nodeutil.ShouldAgentShutdown(err) {
				return nil, err
			}
```